### PR TITLE
Dpdk 17.05

### PR DESCRIPTION
**Pull request by outscale-mgo**

### CODE REVIEW

File: src/packetsgen.c [hunk 1]
```diff
@@ -76,6 +76,7 @@ static int packetsgen_poll(struct pg_brick *brick,
 	pkts = g_new0(struct rte_mbuf*, state->packets_nb);
 	for (i = 0; i < state->packets_nb; i++) {
 		pkts[i] = rte_pktmbuf_clone(state->packets[i], mp);
+		g_assert(pkts[i]);
 		pkts[i]->udata64 = i;
 	}
```
Comment (outscale-jju) [inline src/packetsgen.c]: paxketgen fail-S

File: src/packetsgen.c [hunk 2]
```diff
@@ -73,7 +73,7 @@ static int packetsgen_poll(struct pg_brick *brick,
 	s = &brick->sides[state->output];
 
 
-	pkts = g_new0(struct rte_mbuf*, state->packets_nb);
```
Comment (outscale-jju) [inline src/packetsgen.c]: packetgen: don't initilaze packet array to 0 

--> Initialize

File: src/ip-fragment.c [hunk 3]
```diff
@@ -187,8 +188,8 @@ static int ip_fragment_burst(struct pg_brick *brick, enum pg_side from,
 					pkts_mask ^= (ONE64 << i);
 			}
 		}
-	} else if (do_reassemble(state, pkts, s, from, &pkts_mask, errp) < 0) {
-		return -1;
+	} else {
+		return do_reassemble(state, pkts, s, from, &pkts_mask, errp);
```
Comment (outscale-jju) [inline src/ip-fragment.c]: TODO

File: tests/antispoof/tests.c [hunk 4]
```diff
@@ -187,12 +187,12 @@ static void test_antispoof_generic(const unsigned char **pkts,
...
-		filtered_pkts = pg_brick_west_burst_get(col_east, \
+		filtered_pkts = pg_brick_west_burst_get(col_east,	\
 							&filtered_pkts_mask, \
 							&error); \
 		g_assert(!error); \
 		g_assert(pg_mask_count(filtered_pkts_mask) == (pass)); \
-		pg_packets_free(filtered_pkts, filtered_pkts_mask); \
+		g_assert(!!filtered_pkts == pass);			       \
 		rte_pktmbuf_free(packet); \
 	}
```
Comment (outscale-jju) [inline tests/antispoof/tests.c]: `antispoof double free, no more !`
->
`antispoof: fix double free`

File: error_print [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    if [ $has_ponysay -eq 0 ]; then
+	ponysay --pony derpysad $1
+    elif [ $has_cowsay -eq 255 ]; then
+	echo $1 | cowsay -d
+    else
+	tput setaf 1
+	echo $1
+	tput sgr0
+    fi
+}
```
Comment (outscale-jju) [inline error_print]: Must check that works without having cowsay or poneysay

File: dpdk_symbols.sh [hunk 6]
```diff
@@ -1,7 +1,9 @@
 #!/bin/bash
 set -e
 
-[ -z "$RTE_SDK" ] && echo "RTE_SDK path is not set" && exit 1
+source error_print
```
Comment (outscale-jju) [inline dpdk_symbols.sh]: error_print -> error_print.sh

File: README.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ split on demand to be run on different core or even merged.
 
 If you want a graphical representation of a graph, you can generate a [dot](https://en.wikipedia.org/wiki/DOT_%28graph_description_language%29) output.
 
-![packetgraph features](https://osu.eu-west-2.outscale.com/jerome.jutteau/16d1bc0517de5c95aa076a0584b43af6/packetgraph_features.png "packetgraph features")
+![packetgraph features](https://osu.eu-west-2.outscale.com/jerome.jutteau/16d1bc051de5c95aa076a0584b43af6/packetgraph_features.png "packetgraph features")
```
Comment (outscale-jju) [inline README.md]: a `7` has been accidentally removed

File: README.md [hunk 8]
```diff
@@ -39,7 +39,7 @@ If you want a graphical representation of a graph, you can generate a [dot](http
 - vtep: VXLAN Virtual Terminal End Point switching packets on virtual LANs, can encapsulate packets over ipv4 or ipv6
 - queue: temporally store packets between graph
 - pmtud(ipv4 only): Path MTU Discovery is an implementation of [RFC 1191](https://tools.ietf.org/html/rfc1191)
-- fragment-ip: fragment and reassemble packets
+- fragment-ip(ipv4 only): fragment and reassemble packets, currently broken
```
Comment (outscale-jju) [inline README.md]: You can add a link to the corresponding issue

File: src/vhost.c [hunk 9]
```diff
@@ -208,13 +216,26 @@ static void vhost_create_socket(struct pg_vhost_state *state, uint64_t flags,
...
+
+
+	ret = rte_vhost_driver_callback_register(path, &virtio_net_device_ops);
+	if (ret) {
+		*errp = pg_error_new_errno(-ret,
+			"Failed to register vhost-user callbacks");
+		goto free_exit;
+	}
 
+	rte_vhost_driver_start(path);
```
Comment (outscale-mgo) [inline src/vhost.c]: yeah a socket

File: include/packetgraph/vhost.h [hunk 10]
```diff
@@ -85,31 +85,45 @@ int pg_vhost_start(const char *base_dir, struct pg_error **errp);
...
 int pg_vhost_enable(uint64_t feature_mask);
 
 /**
- * Disable virtio features.
+ * Enable virtio features on a brick.
  * @param   feature_mask virtio feature (see VIRTIO_NET_F_* in linux/virtio_net.h)
+ * @param   brick
+ * @return  0 on success, -1 on error
+ */
+int pg_vhost_brick_enable(struct pg_brick *brick, uint64_t feature_mask);
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: -brick

File: include/packetgraph/vhost.h [hunk 11]
```diff
@@ -85,31 +85,45 @@ int pg_vhost_start(const char *base_dir, struct pg_error **errp);
...
  */
 void pg_vhost_stop(void);
 
 /**
- * Enable virtio features.
+ * Enable virtio features, every vhost brick will inherit from this
+ * @param   feature_mask virtio feature (see VIRTIO_NET_F_* in linux/virtio_net.h)
+ * @return  0 on success, -1 on error
+ */
+int pg_vhost_disable(uint64_t feature_mask);
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: %s/pg_vhost_disable/pg_vhost_disable_global/

File: include/packetgraph/vhost.h [hunk 12]
```diff
@@ -85,31 +85,45 @@ int pg_vhost_start(const char *base_dir, struct pg_error **errp);
...
+
+/**
+ * Enable virtio features, every vhost brick will inherit from this
  * @param   feature_mask virtio feature (see VIRTIO_NET_F_* in linux/virtio_net.h)
  * @return  0 on success, -1 on error
  */
 int pg_vhost_enable(uint64_t feature_mask);
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: %s/pg_vhost_enable/pg_vhost_enable_global/

File: src/packetsgen.c (hunk 1 above)
Comment (outscale-jju) [inline src/packetsgen.c]: paxketgen in commit

File: src/utils/mempool.h [hunk 13]
```diff
@@ -27,7 +27,10 @@
 
 extern struct rte_mempool *mp;
 
-void pg_alloc_mempool(void);
+/**
+ * @param falgs mempool flags
```
Comment (outscale-jju) [inline src/utils/mempool.h]: falgs

**Comment by outscale-jju** [comment]
merged

**Pull request closed without merging**